src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -473,5 +496,143 @@ private ImmutableArray<AsyncCompletionData.CompletionFilter> GetFilters(RoslynCo
 
             return current == questionPosition;
         }
+
+        /// <summary>
+        /// Provides an efficient way to compute a set of completion filters associated with a collection of completion items.
+        /// Presence of expander and filter in the set have different meanings. Set contains a filter means the filter is
+        /// available but unselected, whereas it means available and selected for an expander. Note that even though VS support 
+        /// having multiple expanders, we only support one.
+        /// </summary>
+        private class FilterSet
+        {
+            // Cache all the VS completion filters which essentially make them singletons.
+            // Because all items that should be filtered using the same filter button must 
+            // use the same reference to the instance of CompletionFilter.
+            private static readonly Dictionary<string, AsyncCompletionData.CompletionFilter> s_filterCache =
+                new Dictionary<string, AsyncCompletionData.CompletionFilter>();
+
+            private static readonly ImmutableArray<int> s_filterMasks;
+            private static readonly int s_expanderMask;
+
+            private BitVector32 _vector;
+
+            public static ImmutableArray<CompletionItemFilter> Filters => CompletionItemFilter.AllFilters;
+
+            private static AsyncCompletionData.CompletionExpander _expander = null;
+            public static AsyncCompletionData.CompletionExpander Expander
+            {
+                get
+                {
+                    if (_expander == null)
+                    {
+                        var addImageId = Shared.Extensions.GlyphExtensions.GetKnownImageId(KnownImageIds.ExpandScope);
+                        _expander = new AsyncCompletionData.CompletionExpander(
+                            EditorFeaturesResources.Expander_display_text,
+                            accessKey: "x",
```

Comment:
>"x" [](start = 39, length = 3)

loc? resource?

Reply:
I don't think accessKey needs to be localized. @dpoeschl?

Reply:
[CompletionItemFilter](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Completion/CompletionItemFilter.cs,65) also contains a bunch of completion filter shortcuts that aren't localized. This should be in the same boat as those, but I'm not sure whether that boat is seaworthy. @tmeschter ?

Reply:
You don't need to worry about that. The only time it impacts localization is when the text for a label (for example) embeds the accelerator key--in which case the translators need to force the same character to appear in the translated resource. That doesn't apply here.